core/services/workflows/store_db.go

```diff
@@ -0,0 +1,263 @@
+package workflows
+
+import (
+	"context"
+	"errors"
+	"fmt"
+
+	"google.golang.org/protobuf/proto"
+
+	"github.com/jmoiron/sqlx"
+
+	"github.com/smartcontractkit/chainlink-common/pkg/sqlutil"
+	"github.com/smartcontractkit/chainlink-common/pkg/values"
+	valuespb "github.com/smartcontractkit/chainlink-common/pkg/values/pb"
+)
+
+// `dbStore` is a postgres-baked
+// data store that persists workflow progress.
+type dbStore struct {
+	db sqlutil.DataSource
```

Comment:
Bit of a long shot but, if this is easily modellable via a in-memory map, what stops us from using a kv abstraction like https://github.com/philippgille/gokv instead? This would make it really easy to keep both implementations up to date.

Reply:
I can see two reasons :)
* In the long-run we'll probably want to evolve this API further to reduce the in-memory working set: that means indices will become important to support more complex query patterns.
* Familiarity/paved road: we use db sqlutil.DataSource elsewhere and are familiar with it.